docs/topics/exceptions.md

```diff
@@ -1,122 +1,706 @@
+
+    // This line is not printed if an exception is thrown
+    println("End of the program")
+//sampleEnd
 }
 ```
+{kotlin-runnable="true"}
 
-And that's not good. Just take a look at [Effective Java, 3rd Edition](https://www.oracle.com/technetwork/java/effectivejava-136174.html), Item 77: *Don't ignore exceptions*.
+As you can see, the `finally` block guarantees that the resource is closed, regardless of whether an exception occurs.
 
-Bruce Eckel says this about checked exceptions:
+In Kotlin, you have the flexibility to use only a `catch` block, only a `finally` block, or both, depending on your 
+specific needs, but a `try` block must always be accompanied by at least one `catch` block or a `finally` block.
 
-> Examination of small programs leads to the conclusion that requiring exception specifications
->could both enhance developer productivity and enhance code quality, but experience with large software projects suggests
->a different result – decreased productivity and little or no increase in code quality.
+## Creating custom exceptions
 
-And here are some additional thoughts on the matter:
+In Kotlin, you can define custom exceptions by creating classes that extend the built-in `Exception` class, allowing you
+to create more specific error types tailored to the unique needs of your application.
 
-* [Java's checked exceptions were a mistake](https://radio-weblogs.com/0122027/stories/2003/04/01/JavasCheckedExceptionsWereAMistake.html) (Rod Waldhoff)
-* [The Trouble with Checked Exceptions](https://www.artima.com/intv/handcuffs.html) (Anders Hejlsberg)
+To create one, you can define a class that extends `Exception`:
 
-If you want to alert callers about possible exceptions when calling Kotlin code from Java, Swift, or Objective-C,
-you can use the `@Throws` annotation. Read more about using this annotation [for Java](java-to-kotlin-interop.md#checked-exceptions)
-and [for Swift and Objective-C](native-objc-interop.md#errors-and-exceptions).
+```kotlin
+class MyException: Exception("My message")
+```
 
-## The Nothing type
+In this example, there is a default error message, "My message", but you can leave it blank if you want.
+
+Custom exceptions can also be a subclass of any pre-existent exception subclass, like the `ArithmeticException` subclass:
 
-`throw` is an expression in Kotlin, so you can use it, for example, as part of an Elvis expression:
+```kotlin
+class NumberTooLargeException: ArithmeticException("My message")
+```
+
+> Exceptions in Kotlin are stateful objects, carrying information specific to the context of their creation, referred to as the [stack trace](#stack-trace).
+> Avoid creating exceptions using `object` declarations. 
+> Instead, create a new instance of the exception every time you need one. 
+> This way, you can ensure the exception's state accurately reflects the specific context.
+> 
+{type="tip"}
+
+> In Kotlin, [classes are final by default](inheritance.md) and cannot be subclassed unless the parent
```

Comment:
Both relate to subclassing and extending exception classes, but I see what you mean!
I restructured the note and moved things around to flow better: 


To create one, you can define a class that extends `Exception`:

```kotlin
class MyException: Exception("My message")
```

In this example, there is a default error message, "My message", but you can leave it blank if you want.

> Exceptions in Kotlin are stateful objects, carrying information specific to the context of their creation, referred to as the [stack trace](#stack-trace).
> Avoid creating exceptions using `object` declarations.
> Instead, create a new instance of the exception every time you need one.
> This way, you can ensure the exception's state accurately reflects the specific context.
>
{type="tip"}

Custom exceptions can also be a subclass of any pre-existent exception subclass, like the `ArithmeticException` subclass:

```kotlin
class NumberTooLargeException: ArithmeticException("My message")
```

> If you want to create subclasses of custom exceptions, you must declare the parent class as `open`
> because [classes are final by default](inheritance.md) and cannot be subclassed otherwise.
> 
> For example:
>
> ```kotlin
> // Declares a custom exception as an open class, making it subclassable
> open class MyCustomException(message: String): Exception(message)
>
> // Creates a subclass of the custom exception
> class SpecificCustomException: MyCustomException("Specific error message")
> ```
>
{type="note"}

Reply:
LGTM!